app/main.py

```diff
@@ -1,34 +1,65 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
         # Create decks and save them to a list `self.decks`
-        pass
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
+        x_start, x_end = sorted([start[0], end[0]])
+        y_start, y_end = sorted([start[1], end[1]])
+
+        self.decks = [
+            Deck(x, y)
+            for x in range(x_start, x_end + 1)
+            for y in range(y_start, y_end + 1)
+        ]
+
+    def get_deck(self, row: int, column: int) -> Deck | None:
         # Find the corresponding deck in the list
-        pass
+        for deck in self.decks:
+            if row == deck.row and column == deck.column:
+                return deck
 
-    def fire(self, row, column):
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        self.get_deck(row, column).is_alive = False
```

Comment:
check that self.get_deck(row, column) returns not None before to set = False
```suggestion
        deck = self.get_deck(row, column)
        if deck is not None:
             deck.is_alive = False
```